### PR TITLE
[AMORO-2982] UI optimization when no available optimizer groups

### LINKED ISSUE #2982: [Improvement]: UI optimization when no available optimizer groups
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Now we visit the `Catalogs` page and  create a new catalog, it's possible to save the form without any existing optimizer groups, since there is no available optimizer groups. In this case, we have to give 
up what we have inputed and then create a new optimizer group first. The UE is terrible for us.

### How should we improve?

If no available optimizer groups, redirect to `Optimizer Groups` page, and then click [Add Group] button.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by tcodehuber**
## Why are the changes needed?

Close #2982.

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by tcodehuber** [comment]
FireShot Capture 001 - Apache Amoro™ Dashboard - 127 0 0 1

File: amoro-ams/amoro-ams-dashboard/src/views/catalogs/Detail.vue [hunk 1]
```diff
@@ -196,7 +196,23 @@ function initData() {
...
-  optimizerGroupList.value = list
+
+  if (list.length === 0) {
+    Modal.confirm({
+      title: t('noResourceGroupsTitle'),
+      content: t('noResourceGroupsContent'),
+      okText: t('goToButtonText'),
+      onOk: async () => {
+        try {
+          window.location.href = '/optimizing?tab=optimizergroup'
```
[the remark this answers is not in the document]
Comment (tcodehuber) [reply amoro-ams/amoro-ams-dashboard/src/views/catalogs/Detail.vue]: > Is it better to use the method provided by `vue router` to do the operation here?

@chouchouji  Thanks for your suggestion,  I have modified it, PTAL.

**Review by chouchouji — APPROVED** [review]

File: amoro-ams/amoro-ams-dashboard/src/views/catalogs/Detail.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (chouchouji) [reply amoro-ams/amoro-ams-dashboard/src/views/catalogs/Detail.vue]: LGTM

**Review by zhangmo8 — APPROVED** [review]
LGTM

**Review by czy006 — APPROVED** [review]
LGTM

**Pull request merged**